### PR TITLE
feat: refactor with authorization, change current account type

### PR DESCRIPTION
**Pull request by pyphilia**
In this PR:
- withAuthorization -> SignedInWrapper (Component)
- ItemLoginWrapper (Component)
- PreventGuestWrapper (Component) - new
- update `currentMember` type

We can discuss the naming.

### CODE REVIEW

File: src/itemLogin/ItemLoginAuthorization.stories.tsx [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+export const LogInForm: Story = {
+  args: {
+    itemLoginSchemaType: ItemLoginSchemaType.Username,
+  },
+  play: async ({ canvasElement }) => {
+    const canvas = within(canvasElement);
+
+    expect(canvas.getByText('Sign In')).toBeVisible();
+  },
+};
```
Comment (spaenleh) [inline src/itemLogin/ItemLoginAuthorization.stories.tsx:52]: This is ok, but I prefer to use the following style: 

```suggestion
export const LogInForm = {
  args: {
    itemLoginSchemaType: ItemLoginSchemaType.Username,
  },
  play: async ({ canvasElement }) => {
    const canvas = within(canvasElement);

    expect(canvas.getByText('Sign In')).toBeVisible();
  },
} satisfies Story;
```

**Review by spaenleh — APPROVED** [review]
Looks good ! I have left a few comments.

File: src/Authorization/PreventGuestWrapper.tsx [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+  currentAccount,
+  error,
+  id,
+  onButtonClick,
+  startIcon = <ClipboardPen />,
+  errorText = 'An error occured.',
+  text = 'You are currently using Graasp with a guest account. In order to use all features of Graasp, you have to log out and create a Graasp account.',
+}: Props): ReactNode => {
+  if (currentAccount) {
+    // guest - should not have access to home
```
Comment (ReidyT) [inline src/Authorization/PreventGuestWrapper.tsx:44]:
```suggestion
    // guest - should not have access to children component
```

File: src/Authorization/PreventGuestWrapper.tsx [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+  error?: JSX.Element;
+  errorText?: string;
+  id?: string;
+  onButtonClick?: () => void;
+  startIcon?: JSX.Element;
+  text?: string | JSX.Element;
+};
+
+const PreventGuestWrapper = ({
+  buttonText = 'Log out and Create a Graasp account',
```
Comment (ReidyT) [inline src/Authorization/PreventGuestWrapper.tsx:33]: I'm wondering if it's necessary to have defaults and to not define as required parameters. In any case, we'll have to redefine their to use the correct language right? Having their as optional add a risk to forgetting to define their with the user's language.

File: src/Authorization/SignedInWrapper.tsx [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+
+const SignedInWrapper = ({
+  currentAccount,
+  redirectionLink,
+  onRedirect,
+  children,
+}: SignedInWrapperProps): SignedInWrapperProps['children'] => {
+  const redirectToSignIn = (): void => {
+    if (!redirectionLink) {
+      return console.debug('No link has been set for redirection');
```
Comment (ReidyT) [inline src/Authorization/SignedInWrapper.tsx:22]:
```suggestion
     console.debug('No link has been set for redirection');
     return;
```

File: src/Authorization/PreventGuestWrapper.stories.tsx [hunk 5]
```diff
@@ -0,0 +1,66 @@
...
+      </div>
+    ),
+  },
+};
+
+export default meta;
+
+type Story = StoryObj<typeof PreventGuestWrapper>;
+
+export const Guest: Story = {
```
Comment (ReidyT) [inline src/Authorization/PreventGuestWrapper.stories.tsx:30]: Should we also use `satisfies Story` here to be consistent?

File: src/itemLogin/ItemLoginAuthorization.tsx [hunk 6]
```diff
@@ -19,99 +14,58 @@ import ItemLoginScreen, { SignInPropertiesType } from './ItemLoginScreen.js';
...
+  }
 
-      // the item could be fetched without errors
-      // because the user is signed in and has access
-      // or because the item is public
-      if (item && item.id) {
-        return <ChildComponent />;
-      }
+  // signed out but can sign in with item login
+  if ((!currentAccount || !currentAccount.id) && itemLoginSchemaType) {
```
Comment (ReidyT) [inline src/itemLogin/ItemLoginAuthorization.tsx:53]: This works too right?

```suggestion
  if (!currentAccount?.id) && itemLoginSchemaType) {
```

**Review by ReidyT — DISMISSED** [review]
LGTM 🧑‍🚀 I wonder if we should make the optional parameters of `PreventGuestWrapper` mandatory. What do you think (I elaborate that directly in the code review)?

**Pull request merged**